controllers/progressivesync_controller.go

```diff
@@ -332,6 +332,28 @@ func (r *ProgressiveSyncReconciler) updateStatusWithRetry(ctx context.Context, p
 	return retryErr
 }
 
+func (r *ProgressiveSyncReconciler) addSyncedAtAnnotation(ctx context.Context, app argov1alpha1.Application, stageName string) error {
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		key := client.ObjectKeyFromObject(&app)
+		latest := argov1alpha1.Application{}
+		if err := r.Client.Get(ctx, key, &latest); err != nil {
+			return err
+		}
+		val, ok := app.Annotations[utils.ProgressiveSyncSyncedAtStageKey]
+		if !ok {
+			latest.Annotations = make(map[string]string)
+		}
+		if val != stageName {
```

Comment:
Is it worth logging something that this has happened for any potential debugging purposes later on?